dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBStreamHandler.java

```diff
@@ -0,0 +1,121 @@
+    public static final String ERROR_PROCESSING_DYNAMO_DBEVENT_MESSAGE = "Error processing DynamoDBEvent";
+    public static final String SUCCESS_MESSAGE = "200 OK";
+    private final ElasticSearchRestClient elasticSearchClient;
+
+    /**
+     * Default constructor for DynamoDBStreamHandler.
+     */
+    @JacocoGenerated
+    public DynamoDBStreamHandler() {
+        this(new ElasticSearchRestClient(HttpClient.newHttpClient(), new Environment()));
+    }
+
+    /**
+     * Constructor for DynamoDBStreamHandler for testing.
+     *
+     * @param elasticSearchRestClient elasticSearchRestClient to be injected for testing
+     */
+    @JacocoGenerated
+    public DynamoDBStreamHandler(ElasticSearchRestClient elasticSearchRestClient) {
+        this.elasticSearchClient = elasticSearchRestClient;
+    }
+
+
+    @Override
+    public String handleRequest(DynamodbEvent event, Context context) {
+        try {
+            for (DynamodbEvent.DynamodbStreamRecord streamRecord : event.getRecords()) {
+                switch (streamRecord.getEventName()) {
+                    case "INSERT":
+                    case "MODIFY":
+                        upsertSearchIndex(streamRecord);
+                        break;
+                    case "REMOVE":
+                        removeFromSearchIndex(streamRecord);
+                        break;
+                    default:
+                        throw new RuntimeException("Not a known operation");
+                }
+            }
+        } catch (InterruptedException | URISyntaxException | IOException e) {
+            logger.error(ERROR_PROCESSING_DYNAMO_DBEVENT_MESSAGE, e);
+            throw new RuntimeException(e);
+        }
+
+        return SUCCESS_MESSAGE;
+    }
+
+    private void upsertSearchIndex(DynamodbEvent.DynamodbStreamRecord streamRecord)
+            throws InterruptedException, IOException, URISyntaxException {
+        String identifier = getIdentifierFromStreamRecord(streamRecord);
+        Map<String, AttributeValue> valueMap = streamRecord.getDynamodb().getNewImage();
+        logger.trace("valueMap={}", valueMap.toString());
+
+        Predicate<String> indexfilter = new IndexFilterBuilder()
+                .withIndex(DATE_YEAR)
+                .withIndex(DESCRIPTION_MAIN_TITLE)
+                .withIndex(CONTRIBUTORS_IDENTITY_NAME)
+                .withIndex(PUBLICATION_TYPE)
+                .build();
+
+        UnaryOperator<String> indexMapping = new IndexMapperBuilder()
+                .withIndex(DATE_YEAR, YEAR)
+                .withIndex(DESCRIPTION_MAIN_TITLE, TITLE)
+                .withIndex(CONTRIBUTORS_IDENTITY_NAME, NAME)
+                .withIndex(PUBLICATION_TYPE, TYPE)
+                .build();
+
+        ValueMapFlattener flattener = new ValueMapFlattener.Builder()
+                .withIndexFilter(indexfilter)
+                .withIndexMapping(indexMapping)
+                .withSeparator(".")
```

Comment:
refactor to constant and put it is insider the flattener. There is no place where we put another separator and we should make things as simple as possible

Reply:
Separator is now a constant, the rest is mostly gone